src/client.rs

```diff
@@ -41,20 +43,31 @@ impl Client {
         if !base_url.ends_with('/') {
             base_url += "/"
         }
+        let http = Self::http_client_builder(user_agent)
+            .build()
+            .context("failed to initialize HTTP client")?;
+        let http_no_sni = Self::http_client_builder(user_agent)
```

Comment:
Was falling back to no SNI chosen because reqwest makes it difficult to use a different SNI? Or some other reason?